File: ruby/darts/darts.rb
class Darts
  def initialize(x, y)
    @x, @y = x, y
  end

  def score
    distance = Math.sqrt(@x ** 2 + @y ** 2)

    if distance > 10
      0
    elsif distance > 5
      1
    elsif distance > 1
      5
    else
      10
    end
  end
end

File: java/darts/src/main/java/Darts.java
import java.lang.Math;

class Darts {
    private double _x, _y;

    Darts(double x, double y) {
        _x = x;
        _y = y;
    }

    int score() {
        var distance = Math.sqrt(_x * _x + _y * _y);

        if (distance > 10)
            return 0;
        else if (distance > 5)
            return 1;
        else if (distance > 1)
            return 5;
        else
            return 10;
    }
}
